sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolicExecution/ExplodedGraph.cs

```diff
@@ -326,7 +326,7 @@ private void EnqueueStartNode()
             var initialProgramState = new ProgramState();
             foreach (var parameter in declarationParameters)
             {
-                var sv = new SymbolicValue();
+                var sv = SymbolicValue.Create(parameter.Type);

```

Comment:
Why didn't you keep the creation of all symbolic values through the factory method? IMO it will be error prone to extend the EG after a while because we will forget to use the factory method...